### PR TITLE
ContentPresenter fallback binding to TemplatedParent's Content

### PR DESCRIPTION
**Pull request by carldebilly**
GitHub Issue (If applicable): #857 Implicit template binding of ContentPresenter.Content does not work

## Feature
`<ContentPresenter>` will now - as a fallback when not set - automatically bind to
  `TemplatedParent`'s `Content` when this one is a `ContentControl`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [X] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [X] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [X] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs [hunk 1]
```diff
@@ -724,6 +724,8 @@ protected override void OnLoaded()
 			SynchronizeContentTemplatedParent();
 
 			UpdateBorder();
+
+			SetImplicitContent();
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs]: should this be called before `SetUpdateTemplate()`?

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs [hunk 2]
```diff
@@ -818,6 +821,36 @@ void setBinding(DependencyProperty property, string path)
 			ContentTemplateRoot = textBlock;
 		}
 
+		private void SetImplicitContent()
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs]: the same thing should be done for `ContentTemplate` as well.

File: src/SamplesApp/UITests.Shared/UITests.Shared.projitems [hunk 3]
```diff
@@ -41,10 +41,6 @@
       <SubType>Designer</SubType>
       <Generator>MSBuild:Compile</Generator>
     </Page>
-    <Page Include="$(MSBuildThisFileDirectory)Windows_Devices\BarometerTests.xaml">
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/UITests.Shared.projitems]: This is an error... will be removed in another commit.

Was removed locally because not working on UWP.

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs (hunk 1 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs]: Has been moved to OnTemplatedParentChanged.

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs]: That would be another PR.

File: src/SamplesApp/SamplesApp.UITests/SampleControlUITestBase.cs [hunk 4]
```diff
@@ -68,7 +68,8 @@ public void BeforeEachTest()
 				}
 			}
 
-			_app = AppInitializer.AttachToApp();
+			var app = AppInitializer.AttachToApp();
+			_app = app ?? _app;
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/SampleControlUITestBase.cs]: Not sure to understand this one ?

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ContentPresenter/ContentPresenter_LocalOverride.xaml.cs [hunk 5]
```diff
@@ -21,6 +21,7 @@ public sealed partial class ContentPresenter_LocalOverride : UserControl
     {
         public ContentPresenter_LocalOverride()
         {
+	        FeatureConfiguration.ContentPresenter.UseImplicitContentFromTemplatedParent = false;
```
Comment (dr1rrb) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ContentPresenter/ContentPresenter_LocalOverride.xaml.cs]: Alignement issue 😁

File: src/SamplesApp/SamplesApp.UITests/SampleControlUITestBase.cs (hunk 4 above)
Comment (carldebilly) [reply src/SamplesApp/SamplesApp.UITests/SampleControlUITestBase.cs]: `AppInitializer.AttachToApp()` can return null if the app is already initialized...
